source/docs/software/commandbased/commands-v3/command-compositions-v3.rst

```diff
@@ -0,0 +1,333 @@
+# Command Compositions
+
+Commands v3 supports both **declarative composition** (like Commands v2) and **imperative sequencing** (using coroutines). This article covers both approaches and when to use each.
+
+## Declarative Composition
+
+Commands v3 retains v2's declarative composition methods. These create command groups that bundle multiple commands together.
+
+### Sequence
+
+Run commands one after another. The sequence finishes when the last command finishes.
+
+.. code-block:: java
+
+   Command auto = Command.sequence(
+     drivetrain.driveToPose(pose1),
+     arm.moveTo(position),
+     gripper.release()
+   ).withAutomaticName();
+
+Equivalent imperative style:
+
+.. code-block:: java
+
+   Command auto = Command.noRequirements().executing(coroutine -> {
+     coroutine.await(drivetrain.driveToPose(pose1));
+     coroutine.await(arm.moveTo(position));
+     coroutine.await(gripper.release());
+   }).named("Auto Sequence");
```

Comment:
This is slightly different from the above example: with `Command.sequence`, the composition owns _every_ mechanism for the full duration of the sequence, while wrapping in `Command.noRequirements` means that the composition only owns the mechanisms while they're in use by something within the composition (allowing other commands to use those mechanisms while they're not used by the composition).

This would be a 1-to-1 equivalent:

```java
Command auto = Command.requiring(drivetrain, arm, gripper).executing(coroutine -> {
   coroutine.await(drivetrain.driveToPose(pose1));
   coroutine.await(arm.moveTo(position));
   coroutine.await(gripper.release());
}).named("Auto Sequence");
```